ui/src/warp_runner/ui_adapter/mod.rs

```diff
@@ -15,23 +15,42 @@ use std::collections::{HashMap, VecDeque};
 use warp::{
     crypto::DID,
     error::Error,
+    logging::tracing::log,
+    multipass::identity::Identity,
     raygun::{self, Conversation, MessageOptions},
 };
 
 use crate::state::{self, chats};
 
+// this function is used in response to warp events. assuming that the DID from these events is valid.
+// Warp sends the Identity over. if the Identity has not been received yet, get_identity will fail for
+// a valid DID.
 pub async fn did_to_identity(
     did: &DID,
     account: &super::Account,
 ) -> Result<state::Identity, Error> {
-    account
-        .get_identity(did.clone().into())
-        .await
-        // if Ok, get the first item in the vector. 
-        // if the vector is empty, become Error::IdentityDoesntExist
-        .and_then(|v| v.first().cloned().ok_or(Error::IdentityDoesntExist))
-        // if Ok, convert from warp::Identity to state::Identity
-        .map(state::Identity::from)
+    let identity = match account.get_identity(did.clone().into()).await {
+        Ok(list) => list.first().cloned(),
+        Err(e) => {
+            log::warn!("multipass couldn't find identity {}: {}", did, e);
+            None
+        }
+    };
+    let identity = match identity {
+        Some(id) => id,
+        None => {
+            let mut default: Identity = Default::default();
+            default.set_did_key(did.clone());
+            let did_str = &did.to_string();
+            // warning: assumes DIDs are very long. this can cause a panic if that ever changes
+            let start = &did_str[8..=10];
```

Comment:
Dont think it will change but yea still something to be cautious about. If you want to play extra cautious you could use the following 
```suggestion
            let start = did_str.get(8..=10).ok_or(Error::Other)?;
```
(same can be for `end` too)
That way it doesnt panic if something does change with the length.